src/plugins/intel_gpu/tests/unit/test_cases/fully_connected_gpu_test.cpp

```diff
@@ -2559,14 +2595,26 @@ class fully_connected_gpu_tests: public ::testing::Test {
         ASSERT_EQ(outputs.size(), size_t(1));
         ASSERT_EQ(outputs.begin()->first, "fc_prim");
 
-        auto output_mem = outputs.begin()->second.get_memory();
-        cldnn::mem_lock<ov::float16> output_ptr(output_mem, get_test_stream());
+        if (is_output_fp16) {
+            auto output_mem = outputs.begin()->second.get_memory();
+            cldnn::mem_lock<ov::float16> output_ptr(output_mem, get_test_stream());
 
-        auto ref_output_mem = get_ref_results();
-        cldnn::mem_lock<ov::float16> output_ptr_ref(ref_output_mem, get_test_stream());
+            auto ref_output_mem = get_ref_results();
+            cldnn::mem_lock<ov::float16> output_ptr_ref(ref_output_mem, get_test_stream());
 
-        for (size_t i = 0; i < output_ptr_ref.size(); i++) {
-            EXPECT_NEAR(output_ptr_ref[i], output_ptr[i], 9.0) << "i = " << i;
+            for (size_t i = 0; i < output_ptr_ref.size(); i++) {
+                EXPECT_NEAR(output_ptr_ref[i], output_ptr[i], 9.0) << "i = " << i;
+            }
+        } else {
+            auto output_mem = outputs.begin()->second.get_memory();
+            cldnn::mem_lock<float> output_ptr(output_mem, get_test_stream());
+
+            auto ref_output_mem = get_ref_results();
+            cldnn::mem_lock<float> output_ptr_ref(ref_output_mem, get_test_stream());
+
+            for (size_t i = 0; i < output_ptr_ref.size(); i++) {
+                EXPECT_NEAR(output_ptr_ref[i], output_ptr[i], 9.0) << "i = " << i;
+            }
```

Comment:
sorry for the belated comment, but this pattern repeats 4 times. could you refactor it?